Assets/Scripts/SS3D/Systems/Health/HealthController.cs

```diff
@@ -23,17 +21,28 @@ private void BecomeGhost(GameObject player, GameObject ghost)
         Mind originMind = originEntity.Mind;
 
         ghostEntity.SetMind(originMind);
-        destroyObjects(originEntity, ghostEntity);
+        RpcDestroyObjects(originEntity);
+        RpcUpdateGhostPosition(originEntity, ghostEntity);
+    }
+
+    [ObserversRpc]
+    private void RpcDestroyObjects(Entity originEntity)
+    {
+        GameObject originEntityGameObject = originEntity.gameObject;
+        originEntityGameObject.GetComponent<Hands>().Dispose(true);
+        originEntityGameObject.GetComponent<Inventory>().Dispose(true);
+        originEntityGameObject.GetComponent<InteractionController>().Dispose(true);
+        originEntityGameObject.GetComponent<StaminaController>().Dispose(true);
+        originEntityGameObject.GetComponent<HumanoidController>().Dispose(true);
```

Comment:
I'm fine with this for now. But please add a comment saying it is a temporary solution and it has to be optimized later on using a `TODO:` tag, like:
`TODO: Optimize these GetComponents, this is a temporary solution.`